src/main/java/org/folio/circulation/infrastructure/storage/CirculationPolicyRepository.java

```diff
@@ -91,31 +82,10 @@ protected CirculationPolicyRepository(
         "Unable to apply circulation rules for unknown holding"));
     }
 
-    String loanTypeId = item.determineLoanTypeForItem();
-    String locationId = item.getLocationId();
-    String materialTypeId = item.getMaterialTypeId();
-    String patronGroupId = user.getPatronGroupId();
-
-    MultiMap params = MultiMap.caseInsensitiveMultiMap();
-    params.add(ITEM_TYPE_ID_NAME, item.getMaterialTypeId());
-    params.add(LOAN_TYPE_ID_NAME, item.determineLoanTypeForItem());
-    params.add(PATRON_TYPE_ID_NAME, user.getPatronGroupId());
-    params.add(LOCATION_ID_NAME, item.getLocationId());
-
-    log.info(
-      "Applying circulation rules for material type: {}, patron group: {}, loan type: {}, location: {}",
-      materialTypeId, patronGroupId, loanTypeId, locationId);
-
-    final CompletableFuture<Result<Drools>> droolsResponse = clients.getCirculationDrools();
-    final CompletableFuture<Result<CirculationRuleMatch>> circulationRulesResponse = FetchSingleRecord.<Location>forRecord("location")
-    .using(locationsStorageClient)
-    .mapTo(Location::from)
-    .whenNotFound(failed(new ServerErrorFailure("Can`t find location")))
-    .fetch(locationId)
-    .thenCombine(droolsResponse, (locationResult, droolsResult) ->
-      locationResult.combine(droolsResult, (location, drools) ->
-        getPolicyAndMatch(drools, params, location)));
-    return circulationRulesResponse;
+    final var applyCondition = forItem(item, user);
+
+    log.info("Applying circulation rules for conditions: {}", applyCondition);
+    return getPolicyAndMatch(applyCondition);
```

Comment:
Personally, I would inline this variable, I think it reads better without it. If we really want the logging, maybe that could be moved into the `getPolicyAndMatch` function.

```suggestion
    return getPolicyAndMatch(forItem(item, user));
```